### PR TITLE
I added codecs everywhere

### PR DESCRIPTION
**Pull request by Kroppeb**
I deviated in a few places to add fields based on the codec, or name helper methods.

### CODE REVIEW

File: mappings/net/minecraft/entity/SpawnGroup.mapping [hunk 1]
```diff
@@ -12,6 +13,7 @@ CLASS net/minecraft/class_1311 net/minecraft/entity/SpawnGroup
 		ARG 5 peaceful
 	METHOD method_27919 getImmediateDespawnRange ()I
 	METHOD method_27920 getDespawnStartRange ()I
+	METHOD method_28307 getByName (Ljava/lang/String;)Lnet/minecraft/class_1311;
```
Comment (liach) [inline mappings/net/minecraft/entity/SpawnGroup.mapping:16]:
```suggestion
	METHOD method_28307 byName (Ljava/lang/String;)Lnet/minecraft/class_1311;
```

File: mappings/net/minecraft/structure/processor/StructureProcessorType.mapping [hunk 2]
```diff
@@ -1,3 +1,6 @@
 CLASS net/minecraft/class_3828 net/minecraft/structure/processor/StructureProcessorType
+	FIELD field_25013 CODEC Lcom/mojang/serialization/Codec;
+	METHOD codec ()Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/structure/processor/StructureProcessorType.mapping]:
```suggestion
	METHOD getCodec ()Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/structure/pool/StructurePoolElementType.mapping [hunk 3]
```diff
@@ -1 +1,2 @@
 CLASS net/minecraft/class_3816 net/minecraft/structure/pool/StructurePoolElementType
+	METHOD codec ()Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/structure/pool/StructurePoolElementType.mapping]:
```suggestion
	METHOD getCodec ()Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/util/StringIdentifiable.mapping [hunk 4]
```diff
@@ -1,2 +1,11 @@
 CLASS net/minecraft/class_3542 net/minecraft/util/StringIdentifiable
 	METHOD method_15434 asString ()Ljava/lang/String;
+	METHOD method_28140 enumCodec (Ljava/util/function/Supplier;Ljava/util/function/Function;)Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/util/StringIdentifiable.mapping]:
```suggestion
	METHOD method_28140 createCodec (Ljava/util/function/Supplier;Ljava/util/function/Function;)Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/util/StringIdentifiable.mapping [hunk 5]
```diff
@@ -1,2 +1,11 @@
...
+		ARG 0 valueSupplier
+		ARG 1 parser
+	METHOD method_28141 enumCodec (Ljava/util/function/ToIntFunction;Ljava/util/function/IntFunction;Ljava/util/function/Function;)Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/util/StringIdentifiable.mapping]:
```suggestion
	METHOD method_28141 createCodec (Ljava/util/function/ToIntFunction;Ljava/util/function/IntFunction;Ljava/util/function/Function;)Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/util/collection/WeightedList.mapping [hunk 6]
```diff
@@ -10,11 +10,18 @@ CLASS net/minecraft/class_4131 net/minecraft/util/collection/WeightedList
 		ARG 1 random
 	METHOD method_23337 pickRandom (Ljava/util/Random;)Ljava/lang/Object;
 		ARG 1 random
+	METHOD method_28338 weightedListCodecOf (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/util/collection/WeightedList.mapping]: If this is a codec for the list should be
```suggestion
	METHOD method_28338 createCodec (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/util/collection/WeightedList.mapping [hunk 7]
```diff
@@ -10,11 +10,18 @@ CLASS net/minecraft/class_4131 net/minecraft/util/collection/WeightedList
...
 		FIELD field_18401 weight I
 		FIELD field_18402 shuffledOrder D
+		METHOD <init> (Ljava/lang/Object;I)V
+			ARG 1 data
+			ARG 2 weight
 		METHOD method_19033 getShuffledOrder ()D
 		METHOD method_19034 setShuffledOrder (F)V
 			ARG 1 random
 		METHOD method_19035 getElement ()Ljava/lang/Object;
+		METHOD method_28341 entryCodecOf (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/util/collection/WeightedList.mapping]:
```suggestion
		METHOD method_28341 createCodec (Lcom/mojang/serialization/Codec;)Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/gen/chunk/NoiseSamplingConfig.mapping [hunk 8]
```diff
@@ -4,6 +4,7 @@ CLASS net/minecraft/class_5308 net/minecraft/world/gen/chunk/NoiseSamplingConfig
 	FIELD field_24801 yScale D
 	FIELD field_24802 xzFactor D
 	FIELD field_24803 yFactor D
+	FIELD field_25188 DOUBLE_CODEC Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/chunk/NoiseSamplingConfig.mapping:7]: If this is scale/factor codec name it `SCALE_CODEC` or `FACTOR_CODEC`. Why does this use a custom double codec tho

File: mappings/net/minecraft/world/gen/decorator/Decorator.mapping [hunk 9]
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/class_3284 net/minecraft/world/gen/decorator/Decorator
+	FIELD field_24983 codec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/decorator/Decorator.mapping:2]: not a static final field I assume?

File: mappings/net/minecraft/world/gen/feature/Feature.mapping [hunk 10]
```diff
@@ -1,6 +1,7 @@
 CLASS net/minecraft/class_3031 net/minecraft/world/gen/feature/Feature
 	FIELD field_13526 BONUS_CHEST Lnet/minecraft/class_2953;
 	FIELD field_21219 FLOWER Lnet/minecraft/class_4624;
+	FIELD field_24837 codec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/feature/Feature.mapping:4]: guess no static final?

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping [hunk 11]
```diff
@@ -42,6 +43,8 @@ CLASS net/minecraft/class_4647 net/minecraft/world/gen/foliage/FoliagePlacer
 		ARG 6 foliageHeight
 		ARG 7 radius
 		ARG 8 leaves
+	METHOD method_28843 getType ()Lnet/minecraft/class_4648;
+	METHOD method_28846 makeCodec (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P4;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:47]: This looks more like filling 4 fields out than creating a codec

File: mappings/net/minecraft/world/gen/foliage/BlobFoliagePlacer.mapping [hunk 12]
```diff
@@ -1,3 +1,15 @@
 CLASS net/minecraft/class_4646 net/minecraft/world/gen/foliage/BlobFoliagePlacer
 	FIELD field_23752 height I
 	FIELD field_24927 CODEC Lcom/mojang/serialization/Codec;
+	METHOD <init> (IIIII)V
+		ARG 1 radius
+		ARG 2 randomRadus
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/foliage/BlobFoliagePlacer.mapping]:
```suggestion
		ARG 2 randomRadius
```

File: mappings/net/minecraft/world/gen/foliage/SpruceFoliagePlacer.mapping [hunk 13]
```diff
@@ -1,3 +1,11 @@
 CLASS net/minecraft/class_4650 net/minecraft/world/gen/foliage/SpruceFoliagePlacer
 	FIELD field_23757 trunkHeight I
 	FIELD field_23758 randomTrunkHeight I
+	FIELD field_24936 CODEC Lcom/mojang/serialization/Codec;
+	METHOD <init> (IIIIII)V
+		ARG 1 radius
+		ARG 2 randomRadius
+		ARG 3 offset
+		ARG 4 randomOffset
+		ARG 5 trunkheight
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/foliage/SpruceFoliagePlacer.mapping]:
```suggestion
		ARG 5 trunkHeight
```

File: mappings/net/minecraft/world/gen/placer/BlockPlacer.mapping [hunk 14]
```diff
@@ -1 +1,3 @@
 CLASS net/minecraft/class_4629 net/minecraft/world/gen/placer/BlockPlacer
+	FIELD field_24865 CODEC Lcom/mojang/serialization/Codec;
+	METHOD method_28673 getCodec ()Lnet/minecraft/class_4630;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/placer/BlockPlacer.mapping]:
```suggestion
	METHOD method_28673 getType ()Lnet/minecraft/class_4630;
```

File: mappings/net/minecraft/world/gen/stateprovider/PillarBlockStateProvider.mapping [hunk 15]
```diff
@@ -1,2 +1,3 @@
 CLASS net/minecraft/class_4655 net/minecraft/world/gen/stateprovider/PillarBlockStateProvider
-	FIELD field_21313 block Lnet/minecraft/class_2248;
+	FIELD field_21313 state Lnet/minecraft/class_2248;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/stateprovider/PillarBlockStateProvider.mapping]:
```suggestion
	FIELD field_21313 block Lnet/minecraft/class_2248;
```

File: mappings/net/minecraft/world/gen/feature/ConfiguredStructureFeature.mapping [hunk 16]
```diff
@@ -1 +1,6 @@
 CLASS net/minecraft/class_5312 net/minecraft/world/gen/feature/ConfiguredStructureFeature
+	FIELD field_24834 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24835 structureFeature Lnet/minecraft/class_3195;
+	FIELD field_24836 featureConfig Lnet/minecraft/class_3037;
+	METHOD <init> (Lnet/minecraft/class_3195;Lnet/minecraft/class_3037;)V
+		ARG 1 arg
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/feature/ConfiguredStructureFeature.mapping]:
```suggestion
		ARG 1 structureFeature
		ARG 2 featureConfig
```

File: mappings/net/minecraft/world/gen/decorator/BeehiveTreeDecorator.mapping [hunk 17]
```diff
@@ -1,2 +1,5 @@
 CLASS net/minecraft/class_4659 net/minecraft/world/gen/decorator/BeehiveTreeDecorator
-	FIELD field_21317 chance F
+	FIELD field_21317 probability F
+	FIELD field_24958 CODEC Lcom/mojang/serialization/Codec;
+	METHOD <init> (F)V
+		ARG 1 chance
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/decorator/BeehiveTreeDecorator.mapping]:
```suggestion
		ARG 1 probability
```

File: mappings/net/minecraft/world/gen/chunk/FlatChunkGenerator.mapping [hunk 18]
```diff
@@ -1,6 +1,8 @@
 CLASS net/minecraft/class_2897 net/minecraft/world/gen/chunk/FlatChunkGenerator
 	FIELD field_24510 generatorConfig Lnet/minecraft/class_3232;
+	FIELD field_24769 CODEC Lcom/mojang/serialization/Codec;
 	METHOD <init> (Lnet/minecraft/class_3232;)V
 		ARG 1 config
 	METHOD method_28002 (Lnet/minecraft/class_2680;)Lnet/minecraft/class_2680;
 		ARG 0 state
+	METHOD method_28545 getSettings ()Lnet/minecraft/class_3232;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/chunk/FlatChunkGenerator.mapping]:
```suggestion
	METHOD method_28545 getGeneratorConfig ()Lnet/minecraft/class_3232;
```

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 19]
```diff
@@ -6,7 +6,9 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
 	FIELD field_13302 alwaysCarvableBlocks Ljava/util/Set;
 	FIELD field_13305 WATER Lnet/minecraft/class_3610;
 	FIELD field_16653 heightLimit I
+	FIELD field_24831 codec Lcom/mojang/serialization/Codec;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping]:
```suggestion
	FIELD field_24831 configCodec Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 20]
```diff
@@ -6,7 +6,9 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
...
 	METHOD <init> (Lcom/mojang/serialization/Codec;I)V
+		ARG 1 codec
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping]:
```suggestion
		ARG 1 configCodec
```

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 21]
```diff
@@ -82,3 +84,4 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
 		ARG 3 scaledRelativeY
 		ARG 5 scaledRelativeZ
 		ARG 7 y
+	METHOD method_28616 getCodec ()Lcom/mojang/serialization/Codec;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping]:
```suggestion
	METHOD method_28616 getConfigCodec ()Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/state/property/Property.mapping [hunk 22]
```diff
@@ -1,4 +1,8 @@
 CLASS net/minecraft/class_2769 net/minecraft/state/property/Property
+	FIELD field_24742 type Ljava/lang/Class;
+	FIELD field_24743 name Ljava/lang/String;
+	FIELD field_24744 hashcode Ljava/lang/Integer;
```
Comment (YanisBft) [inline mappings/net/minecraft/state/property/Property.mapping]:
```suggestion
	FIELD field_24744 hashCode Ljava/lang/Integer;
```

**Review by YanisBft — COMMENTED** [review]
Round one of reviews, too many files :concern:

File: mappings/net/minecraft/structure/processor/GravityStructureProcessor.mapping [hunk 23]
```diff
@@ -2,3 +2,6 @@ CLASS net/minecraft/class_3795 net/minecraft/structure/processor/GravityStructur
 	FIELD field_16723 heightmap Lnet/minecraft/class_2902$class_2903;
 	FIELD field_16725 offset I
 	FIELD field_25002 CODEC Lcom/mojang/serialization/Codec;
+	METHOD <init> (Lnet/minecraft/class_2902$class_2903;I)V
+		ARG 1 type
```
Comment (YanisBft) [inline mappings/net/minecraft/structure/processor/GravityStructureProcessor.mapping:6]:
```suggestion
		ARG 1 heightmap
```

File: mappings/net/minecraft/world/gen/stateprovider/PillarBlockStateProvider.mapping (hunk 15 above)
Comment (Juuxel) [reply mappings/net/minecraft/world/gen/stateprovider/PillarBlockStateProvider.mapping]: Yeah, even though this is serialised as `state`, it goes through block<->state conversion when it does that.

File: mappings/net/minecraft/world/gen/chunk/NoiseSamplingConfig.mapping (hunk 8 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/chunk/NoiseSamplingConfig.mapping:7]: It's both used by scale and factor and is a Codec that only allows doubles in the range of `0.001` to `1000` IIRC

File: mappings/net/minecraft/world/gen/decorator/Decorator.mapping (hunk 9 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/decorator/Decorator.mapping:2]: nope

File: mappings/net/minecraft/world/gen/carver/Carver.mapping (hunk 21 above)
Comment (Juuxel) [reply mappings/net/minecraft/world/gen/carver/Carver.mapping]: This is a codec that serialises a configured carver.

File: mappings/net/minecraft/world/gen/feature/Feature.mapping (hunk 10 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/feature/Feature.mapping:4]: Nope, holds a `Codec<ConfiguredFeature<FC,Feature<FC>>>`

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping (hunk 11 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:47]: Yeah, this is how they handle inheritance it seems. I didn't know what name to use.

File: mappings/net/minecraft/world/gen/stateprovider/PillarBlockStateProvider.mapping (hunk 15 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/stateprovider/PillarBlockStateProvider.mapping]: ah, hadn't really noticed that.

File: mappings/net/minecraft/world/gen/carver/Carver.mapping (hunk 21 above)
Comment (Kroppeb) [reply mappings/net/minecraft/world/gen/carver/Carver.mapping]: oops, yeah, went a bit too fast on the suggestions.

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping (hunk 11 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:47]: maybe `fillFields`?

**Comment by liach** [comment]
@Kroppeb hope you can take https://github.com/FabricMC/yarn/pull/1453#discussion_r437797446

File: mappings/net/minecraft/world/gen/chunk/NoiseSamplingConfig.mapping (hunk 8 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/chunk/NoiseSamplingConfig.mapping:7]: then `MULTIPLIER_CODEC` I'd suggest, as both scale and factor are multipliers.
```suggestion
	FIELD field_25188 MULTIPLIER_CODEC Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping (hunk 11 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:47]:
```suggestion
	METHOD method_28846 fillFields (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P4;
```

Comment (liach) [reply mappings/net/minecraft/world/gen/foliage/FoliagePlacer.mapping:47]: or maybe
```suggestion
	METHOD method_28846 fillFoliagePlacerFields (Lcom/mojang/serialization/codecs/RecordCodecBuilder$Instance;)Lcom/mojang/datafixers/Products$P4;
```
like `createXxxAttributes` for living entities.

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 24]
```diff
@@ -6,7 +6,9 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
 	FIELD field_13302 alwaysCarvableBlocks Ljava/util/Set;
 	FIELD field_13305 WATER Lnet/minecraft/class_3610;
 	FIELD field_16653 heightLimit I
+	FIELD field_24831 configCodec Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping:9]:
```suggestion
	FIELD field_24831 configuratedCodec Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 25]
```diff
@@ -6,7 +6,9 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
...
 	METHOD <init> (Lcom/mojang/serialization/Codec;I)V
+		ARG 1 configCodec
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping:11]:
```suggestion
		ARG 1 configuratedCodec
```

File: mappings/net/minecraft/world/gen/carver/Carver.mapping [hunk 26]
```diff
@@ -82,3 +84,4 @@ CLASS net/minecraft/class_2939 net/minecraft/world/gen/carver/Carver
 		ARG 3 scaledRelativeY
 		ARG 5 scaledRelativeZ
 		ARG 7 y
+	METHOD method_28616 getConfigCodec ()Lcom/mojang/serialization/Codec;
```
Comment (liach) [inline mappings/net/minecraft/world/gen/carver/Carver.mapping:87]:
```suggestion
	METHOD method_28616 getConfiguratedCodec ()Lcom/mojang/serialization/Codec;
```

File: mappings/net/minecraft/world/biome/source/CheckerboardBiomeSource.mapping [hunk 27]
```diff
@@ -1,5 +1,8 @@
 CLASS net/minecraft/class_1973 net/minecraft/world/biome/source/CheckerboardBiomeSource
+	FIELD field_24715 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24716 scale I
 	FIELD field_9480 gridSize I
-	FIELD field_9481 biomeArray Ljava/util/List;
+	FIELD field_9481 biomeList Ljava/util/List;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/source/CheckerboardBiomeSource.mapping:5]:
```suggestion
	FIELD field_9481 biomes Ljava/util/List;
```

File: mappings/net/minecraft/world/biome/source/TheEndBiomeSource.mapping [hunk 28]
```diff
@@ -1,4 +1,10 @@
 CLASS net/minecraft/class_2169 net/minecraft/world/biome/source/TheEndBiomeSource
+	FIELD field_24730 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24731 seed J
 	FIELD field_9830 BIOMES Ljava/util/List;
 	FIELD field_9831 noise Lnet/minecraft/class_3541;
+	METHOD <init> (J)V
+		ARG 1 seed
+	METHOD method_28479 isSeed (J)Z
```
Comment (YanisBft) [inline mappings/net/minecraft/world/biome/source/TheEndBiomeSource.mapping:8]:
```suggestion
	METHOD method_28479 isSeedEqual (J)Z
```

File: mappings/net/minecraft/world/gen/chunk/SurfaceChunkGenerator.mapping [hunk 29]
```diff
@@ -13,6 +13,12 @@ CLASS net/minecraft/class_3754 net/minecraft/world/gen/chunk/SurfaceChunkGenerat
 	FIELD field_16581 upperInterpolatedNoise Lnet/minecraft/class_3537;
 	FIELD field_16648 AIR Lnet/minecraft/class_2680;
 	FIELD field_24773 CODEC Lcom/mojang/serialization/Codec;
+	FIELD field_24774 settings Lnet/minecraft/class_5284;
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/chunk/SurfaceChunkGenerator.mapping:16]:
```suggestion
	FIELD field_24774 chunkGeneratorType Lnet/minecraft/class_5284;
```

File: mappings/net/minecraft/world/gen/chunk/SurfaceChunkGenerator.mapping [hunk 30]
```diff
@@ -13,6 +13,12 @@ CLASS net/minecraft/class_3754 net/minecraft/world/gen/chunk/SurfaceChunkGenerat
...
+	FIELD field_24778 seed J
+	FIELD field_24779 maxY I
+	METHOD <init> (Lnet/minecraft/class_1966;Lnet/minecraft/class_1966;JLnet/minecraft/class_5284;)V
+		ARG 3 seed
+		ARG 5 settings
```
Comment (YanisBft) [inline mappings/net/minecraft/world/gen/chunk/SurfaceChunkGenerator.mapping:21]:
```suggestion
		ARG 5 chunkGeneratorType
```

**Comment by YanisBft** [comment]
Now 100% superseded

**Pull request closed without merging**